### PR TITLE
Fixes #1 - Add ConnectFourBoard::is_game_over() that detects ties

### LINKED ISSUE #1: Need to handle ties
The grid can potentially fill up before a winner is determined.

### PR DESCRIPTION
**Pull request by brantonb**

### CODE REVIEW

File: ConnectFour/connect_four_board.py [hunk 1]
```diff
@@ -45,6 +45,19 @@ def winner(self):
...
+            return True
+
+        # Look for any column that isn't full and return False.
+        for column in self._columns:
+            if len(column) < self._height:
+                return False
+
+        # All columns are full, so there is a draw.
+        return True
+
```
Comment (jdharms) [inline ConnectFour/connect_four_board.py]: Alternatively,
````python
    def is_game_over(self):
        return any(len(column) < self._height for column in self.columns)
````
See: https://www.python.org/dev/peps/pep-0289/ and https://docs.python.org/3.5/library/functions.html#any

Yours is fine and I wouldn't leave a comment on coworker's code--but I'm in educational mode.

File: ConnectFour/connect_four_board_tests.py [hunk 2]
```diff
@@ -107,10 +107,34 @@ def test_negative_indices_causes_false_winner(self):
...
+            # RBRB
+            # BRBR
+            if row < 3:
+                color = self._toggle_color(color)
+            for col in range(4):
+                self.assertFalse(board.is_game_over())
+                board.drop_piece(color, col)
+                color = self._toggle_color(color)
+
+        self.assertTrue(board.is_game_over())
```
Comment (jdharms) [inline ConnectFour/connect_four_board_tests.py]: Consider adding similar assertions for board.is_game_over() to a few other existing or new test cases.  (is_game_over() is true when there's a winner, false when there's not yet one.)  Your new game logic relies on it, while your tests just use board.winner()

**Comment by jdharms** [comment]
Looks good to me

**Comment by brantonb** [comment]
Responded to CR feedback. I like the suggestions, thanks!

**Comment by jdharms** [comment]
Oops, I missed a "not", huh?

**Comment by brantonb** [comment]
Just keeping me on my toes. Unit tests are great.

**Pull request merged**